tools/cldr-apps/src/main/java/org/unicode/cldr/web/STFactory.java

```diff
@@ -1209,14 +1193,16 @@ private void saveVoteToDb(final User user, final String distinguishingXpath, fin
                 conn.commit();
             } catch (SQLException e) {
                 SurveyLog.logException(logger, e, "Exception in saveVoteToDb");
-                SurveyMain.busted("Could not vote for value in locale locale " + locale, e);
+                SurveyMain.busted("Could not vote for value in locale " + locale, e);
                 throw new InternalError("Could not load locale " + locale + " : " + DBUtils.unchainSqlException(e));
             } finally {
-                DBUtils.close(saveOld, ps, ps2, conn);
+                DBUtils.close(saveOld, ps, conn);
             }
             SurveyLog.debug(et);
 
-            if (sm.fora != null && !voteIsAutoImported) {
+            // Voting can trigger adding a forum post (agree/decline) and/or closing a forum thread.
+            // AUTO_IMPORT and MANUAL_IMPORT votes are excluded; DIRECT and BULK_UPLOAD are not excluded.
+            if (sm.fora != null && (voteType != VoteType.AUTO_IMPORT && voteType != VoteType.MANUAL_IMPORT)) {
                 sm.fora.doForumAfterVote(locale, user, distinguishingXpath, xpathId, value, didClearFlag);
```

Comment:
The exclusion of MANUAL_IMPORT is new -- I'm not sure whether such forum posts have been made, but they would be very strange given that such votes are done in the name of an anonymous "user", and so the forum posts would also be from anonymous users (whose votes have zero weight)

Reply:
I checked the db -- there has been exactly one forum post by an anonymous user:
```
mysql> select id,poster,loc,xpath,last_time,version,type,is_open from cldr_forum_posts where poster >= 2055 and poster <= 2074 limit 10;
+-------+--------+------+--------+---------------------+---------+------+---------+
| id    | poster | loc  | xpath  | last_time           | version | type | is_open |
+-------+--------+------+--------+---------------------+---------+------+---------+
| 53706 |   2055 | ps   | 687913 | 2020-07-06 04:29:12 | 38      |    3 |       0 |
+-------+--------+------+--------+---------------------+---------+------+---------+
1 row in set (0.01 sec)
```

It's an "Agree" post (type 3).

Reply:
https://st.unicode.org/cldr-apps/v#/ps/EnergyPower/376496946a9c5f0b

Reply:
I don't know how an anonymous user would post.  Almost sounds like an error?